keras_contrib/utils/save_load_utils.py

```diff
@@ -0,0 +1,107 @@
+import warnings
+
+import h5py
+import keras.backend as K
+from keras import optimizers
+from keras.engine import topology
+from keras.legacy import models as legacy_models
+
+
+def save_all_weights(model, filepath, include_optimizer=True):
+    """
+    Save model weights and optimizer weights but not configuration to a HDF5 file.
+    Functionally between `save` and `save_weights`.
+
+    The saved model contains:
+        - the model's weights
+        - the model's optimizer's state (if any)
+    If you have a complicated model or set of models that do not serialize to JSON correctly, use this method.
+    # Arguments
+        model: Keras model instance to be saved.
+        filepath: String, path where to save the model.
+        include_optimizer: If True, save optimizer's state together.
+    # Raises
+        ImportError: if h5py is not available.
+    """
+    if h5py is None:
+        raise ImportError('`save_all_weights` requires h5py.')
+
+    with h5py.File(filepath, 'w') as f:
+        model_weights_group = f.create_group('model_weights')
+        if legacy_models.needs_legacy_support(model):
+            model_layers = legacy_models.legacy_sequential_layers(model)
+        else:
+            model_layers = model.layers
+        topology.save_weights_to_hdf5_group(model_weights_group, model_layers)
```

Comment:
I came across an error in this step. The error is :    RuntimeError: Unable to create attribute (Object header message is too large).

Reply:
@xmengli999 can you post some more details? Which line is actually failing? (Maybe make an issue and link it here)

Reply:
It fails at line 35. "topology.save_weights_to_hdf5_group(model_weights_group, model_layers)". My model has a huge number of layers (around 1600). The error is : RuntimeError: Unable to create attribute (Object header message is too large). I also refer to the solution you provided in the link:
https://github.com/fchollet/keras/issues/6766
But it seems not workable in my case. Is it because my model has too many layers?

Reply:
What is the difference of your function "save_all_weights" with the function "save_model" provided in keras?

Reply:
Any answer here? I thought this code would help me as well but it doesnt. As of now it is not possible tp save NastNet weights without significant hacking. Im still hacking and havent figured it out yet.

Reply:
topology is also deprecated in keras 2.2.0 and onwards. So this needs to be replaced by saving to work. This is noted in [matterport/MASK_RCNN#566](https://github.com/matterport/Mask_RCNN/issues/566)

The fix as suggested in [matterport/MASK_RCNN#662](https://github.com/matterport/Mask_RCNN/pull/662) was to replace topology with saving.

I have also tried this function as this was the solution to [keras-team/keras$6766](https://github.com/keras-team/keras/issues/6766), however, with the fix I mentioned above as I am currently on version 2.2.2. It did not work and results in the same error from the issues and like @xmengli999 

I will look more into the difference between saving and topology to see if the fix is the cause, but it doesn't seem like it.